bootstrap/handlers/externalmqtt.go

```diff
@@ -0,0 +1,171 @@
+
+	brokerConfig := configuration.GetBootstrap().ExternalMQTT
+	topics := brokerConfig.SubscribeTopics
+	// TODO: remove first check and update error log in EdgeX 3.0
+	if len(strings.TrimSpace(topics)) == 0 && len(brokerConfig.Topics) == 0 {
+		lc.Errorf("missing SubscribeTopics and/or Topics for external MQTT connection. Must be present in [MessageQueue.External] section")
+		return false
+	}
+
+	_, err := url.Parse(brokerConfig.Url)
+	if err != nil {
+		lc.Errorf("invalid MQTT Broker Url '%s': %s", brokerConfig.Url, err.Error())
+		return false
+	}
+
+	opts := mqtt.NewClientOptions()
+	opts.AddBroker(brokerConfig.Url)
+	opts.SetClientID(brokerConfig.ClientId)
+	opts.SetOnConnectHandler(e.onConnectHandler)
+	opts.SetAutoReconnect(brokerConfig.AutoReconnect)
+	opts.KeepAlive = brokerConfig.KeepAlive
+	if len(brokerConfig.ConnectTimeout) > 0 {
+		duration, err := time.ParseDuration(brokerConfig.ConnectTimeout)
+		if err != nil {
+			lc.Errorf("invalid MQTT ConnectTimeout '%s': %s", brokerConfig.ConnectTimeout, err.Error())
+			return false
+		}
+		opts.SetConnectTimeout(duration)
+	}
+
+	secretProvider := container.SecretProviderFrom(dic.Get)
+	authMode := brokerConfig.AuthMode
+	if brokerConfig.AuthMode == "" {
+		authMode = messaging.AuthModeNone
+		lc.Warn("AuthMode not set, defaulting to \"" + messaging.AuthModeNone + "\"")
+	}
+
+	//get the secrets from the secret provider and populate the struct
+	secretData, err := messaging.GetSecretData(authMode, brokerConfig.SecretPath, secretProvider)
+	if err != nil {
+		lc.Errorf("Failed to retrieve secret data: %s", err.Error())
+		return false
+	}
+	//ensure that the AuthMode selected has the required secret values
+	if secretData != nil {
+		err = messaging.ValidateSecretData(authMode, brokerConfig.SecretPath, secretData)
+		if err != nil {
+			lc.Errorf("Invalid secret data: %s", err.Error())
+			return false
+		}
+
+		// configure the mqtt client with the retrieved secret values
+		tlsConfig := &tls.Config{
+			// nolint: gosec
+			InsecureSkipVerify: brokerConfig.SkipCertVerify,
+			MinVersion:         tls.VersionTLS12,
+		}
+		switch authMode {
+		case messaging.AuthModeUsernamePassword:
+			opts.SetUsername(secretData.Username)
+			opts.SetPassword(secretData.Password)
+		case messaging.AuthModeCert:
+			cert, err := tls.X509KeyPair(secretData.CertPemBlock, secretData.KeyPemBlock)
+			if err != nil {
+				lc.Errorf("Failed to parse public/private key pair: %s", err.Error())
+				return false
+			}
+			tlsConfig.Certificates = []tls.Certificate{cert}
+		case messaging.AuthModeCA:
+			// Nothing to do here for this option
+		case messaging.AuthModeNone:
+			// Nothing to do here for this option
+		}
+
+		if len(secretData.CaPemBlock) > 0 {
+			caCertPool := x509.NewCertPool()
+			ok := caCertPool.AppendCertsFromPEM(secretData.CaPemBlock)
+			if !ok {
+				//return nil, errors.New("error parsing CA PEM block")
+				return false
+			}
+			tlsConfig.ClientCAs = caCertPool
```

Comment:
```suggestion
			tlsConfig.RootCAs = caCertPool
```
`ClientCAs` defines the set of root certificate authorities that servers use if required to verify a client certificate..
`RootCAs` defines the set of root certificate authorities that clients use when verifying server certificates.